### PR TITLE
First stab at quickwit UI.

### PR DESCRIPTION
**Pull request by fmassot**

### CODE REVIEW

**Review by fulmicoton — APPROVED** [review]
what a beast O_o
I could go in depth but I went through everything rapidly and did not find anything unexpected, and had no trouble navigating through the code. Awesome job.

I think we can merge this already. What do you think Francois?

File: quickwit-ui/app/src/utils/models.ts [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+  numHits: 100,
+}
+
+export type SearchResponse = {
+  count: number;
+  hits: Array<RawDoc>;
+  numMicrosecs: number;
+}
+
+export type IndexMetadata = {
```
Comment (PSeitz) [inline quickwit-ui/app/src/utils/models.ts:137]: we should consider generating these types

https://github.com/Aleph-Alpha/ts-rs

File: quickwit-ui/app/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    "lib": [
+      "dom",
+      "dom.iterable",
+      "esnext"
+    ],
+    "allowJs": true,
+    "skipLibCheck": true,
+    "esModuleInterop": true,
+    "allowSyntheticDefaultImports": true,
+    "strict": true,
```
Comment (PSeitz) [inline quickwit-ui/app/tsconfig.json:13]: strict does not cover all strict options on newer versions automatically

noPropertyAccessFromIndexSignature
noUncheckedIndexedAccess
noUnusedLocals
noUnusedParameters

File: quickwit-ui/app/package.json [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    "typescript": "^4.4.2",
+    "web-vitals": "^2.1.0"
+  },
+  "scripts": {
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "test": "react-scripts test",
+    "eject": "react-scripts eject"
+  },
+  "eslintConfig": {
```
Comment (PSeitz) [inline quickwit-ui/app/package.json:33]: config, but no eslint in the dependencies?

File: quickwit-ui/app/src/utils/models.ts (hunk 1 above)
Comment (fmassot) [reply quickwit-ui/app/src/utils/models.ts:137]: oh nice! Will definitely give it a try. I will put that in a separate issue.

File: quickwit-ui/app/tsconfig.json (hunk 2 above)
Comment (fmassot) [reply quickwit-ui/app/tsconfig.json:13]: Will take a look on that.

**Comment by fmassot** [comment]
> I think we can merge this already. What do you think Francois?

Let me make a final pass on it and we could do it. By the way, I want to remove the subfolder app as the rust code related to the REST endpoints will be in the quickwit-server crate.

File: quickwit-ui/app/tsconfig.json (hunk 2 above)
Comment (fmassot) [reply quickwit-ui/app/tsconfig.json]: I set everything to true

File: quickwit-ui/app/package.json (hunk 3 above)
Comment (fmassot) [reply quickwit-ui/app/package.json]: no ideas, I'm so lazy with the js stack...

Comment (PSeitz) [reply quickwit-ui/app/package.json]: we should add an eslinter

**Pull request merged**